### PR TITLE
adding __await__ magic method to Query

### PR DESCRIPTION
**Pull request by dantownsend**
Following this suggestion from Martin:

https://github.com/piccolo-orm/piccolo/issues/15

### CODE REVIEW

File: piccolo/query/base.py [hunk 1]
```diff
@@ -87,6 +87,13 @@ def _validate(self):
         """
         pass
 
+    def __await__(self):
+        """
+        If the user doesn't explicity call .run(), proxy to it as a
+        convenience.
+        """
+        return self.run().__await__()
```
Comment (uwinx) [inline piccolo/query/base.py:95]: if you want to keep respecting `run`'s `in_pool` flag in `__await__`, you can add `await_in_pool` property flag (setter+getter) to `Query`

Comment (dantownsend) [reply piccolo/query/base.py:95]: @unwinx How do you envisage that working?

Like this, or something else?

```python
query = MyTable.select()
query.in_pool = False
await query
```

**Comment by uwinx** [comment]
exactly. that'd be "just in case user wants X" 18:40, 6 November 2020, Daniel Townsend <notifications@github.com>:
@dantownsend commented on this pull request.



In piccolo/query/base.py:
> @@ -87,6 +87,13 @@ def _validate(self):
         """
         pass
 
+    def __await__(self):
+        """
+        If the user doesn't explicity call .run(), proxy to it as a
+        convenience.
+        """
+        return self.run().__await__()

@unwinx How do you envisage that working?
Like this, or something else?
query = MyTable.select()
query.in_pool = False
await query

File: piccolo/query/base.py (hunk 1 above)
Comment (uwinx) [reply piccolo/query/base.py:95]: yes, exactly. as a "just in case" feature

Comment (dantownsend) [reply piccolo/query/base.py:95]: I'll leave this out for now - people can use `.run()` if they need more control.

**Pull request merged**